src/finch/compiled.py

```diff
@@ -29,6 +34,45 @@ def _to_lazy_tensor(x: Tensor | Any, /) -> Tensor | Any:
     return x if not isinstance(x, Tensor) else lazy(x)
 
 
+def _recurse_iter_return(x: IterObj, /, *, compute_kwargs: dict[str, Any]) -> IterObj:
+    current = 0
+
+    def recursor(_: Any) -> tuple[int, Any]:
+        nonlocal current
+        ret = current
+        current += 1
+        return ret
+
+    # Make a recursive iterator of indices.
+    idx_obj = _recurse(x, f=recursor)
+    jl_computed = []
+    py_computed = []
+
+    # Collect lazy tensors; use placeholder
+    _placeholder = object()
+    for x in _recurse_iter(x):
+        if isinstance(x, Tensor) and not x.is_computed():
+            jl_computed.append(x._obj)
+            py_computed.append(_placeholder)
+        else:
+            py_computed.append(x)
```

Comment:
This path is used when one of the arguments is e.g. just an `int`, right? (non-Tensor things)

Reply:
Yes, I wasn't 100% sure if we could call `jl.Finch.compute` on these things. Anyway, I felt it was better not to.